packages/classic-webview/src/pages/UnlockHardcorePage.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { HardcoreLogo } from '@hotandcold/webview-common/components/logo';
+import { GoldIcon } from '@hotandcold/webview-common/components/icon';
+
+const primaryBorderColor = '#FFBF0B';
+const secondaryBorderColor = '#8BA2AD';
+const secondaryPillBgColor = '#2A3236';
+
+// Inline PurchaseButton component
+interface PurchaseButtonProps {
+  mainText: string;
+  style: 'primary' | 'secondary';
+  badgeIcon: React.ReactNode;
+  badgeText: string;
+  onClick?: () => void;
+}
+
+const PurchaseButton: React.FC<PurchaseButtonProps> = (props) => {
+  const { mainText, badgeIcon, badgeText, onClick, style } = props;
+  const color = style === 'primary' ? primaryBorderColor : secondaryBorderColor;
+  const pillBg = style === 'secondary' ? secondaryPillBgColor : color;
+  const pillTextColor = style === 'secondary' ? 'white' : 'black';
+
+  return (
+    <button
+      onClick={onClick}
+      className="flex w-full flex-auto flex-row items-center justify-between gap-4 whitespace-nowrap rounded-[64px] border-2 p-3 sm:w-auto"
+      style={{ borderColor: color }}
+    >
+      <span
+        className="text-left font-sans text-base font-semibold leading-5 tracking-tight"
+        style={{ color }}
+      >
+        {mainText}
+      </span>
+      <span
+        className="flex h-8 w-fit flex-row items-center gap-1 rounded-full px-3 py-2 text-xs font-semibold"
+        style={{ backgroundColor: pillBg, color: pillTextColor }}
+      >
+        <span className="h-4 w-4">{badgeIcon}</span>
+        <span className="text-center leading-4 tracking-[-0.1px]">{badgeText}</span>
```

Comment:
I don't think you need any of these styles:

```suggestion
        {badgeText}
```